### DISCUSSION TITLE
How to get MikroORM in test file of Nestjs project

### DISCUSSION THREAD

**Discussion by Jedliu** [mikro-orm/mikro-orm#3042] [Q&A] [answered]
Hi All,

I'm building a demo Nestjs project using MikroORM. I'd like to instantiate a factory inside the test but I can't get the orm object. 

Here is the code snippet.
```ts
describe('AppController (e2e)', () => {
  let app: INestApplication;
  let orm: MikroORM;

  beforeAll(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();

    orm = app.get(MikroORM);
    const user = new UserFactory(orm.em).makeOne();
    console.log('user :>> ', user);
  });
```
There is an error when running the test.
```
ValidationError: Using global EntityManager instance methods for context specific actions is disallowed. If you need to work with the global instance's identity map, use `allowGlobalContext` configuration option or `fork()` instead.
```
Here is a repo for your reference [nestjs-mikroorm-demo](https://github.com/Jedliu/nestjs-mikroorm-demo) and here is the way to reproduce the error
```
npm run test:e2e
```

Thank you!

**Comment by B4nan** [comment, marked as answer]
Either fork the EM or allow the global context, for unit tests its fine (as long as you understand what it means - you will need to clear the context yourself to simulate new requests).

```ts
const user = new UserFactory(orm.em.fork()).makeOne();
```

**Reply by Jedliu** [reply]
OMG, I really appreciate your fast reply! 

I've read this part of the doc but I didn't know this could be used here. Thank you so much!

**Reply by Jedliu** [reply]
By the way, here is the doc [forking-entity-manager](https://mikro-orm.io/docs/identity-map#forking-entity-manager).
